dll/win32/browseui/shellfind/CFindFolder.cpp

```diff
@@ -1056,8 +1142,30 @@ STDMETHODIMP CFindFolder::GetDisplayNameOf(PCUITEMID_CHILD pidl, DWORD dwFlags,
 STDMETHODIMP CFindFolder::SetNameOf(HWND hwndOwner, PCUITEMID_CHILD pidl, LPCOLESTR lpName, DWORD dwFlags,
                                     PITEMID_CHILD *pPidlOut)
 {
-    UNIMPLEMENTED;
-    return E_NOTIMPL;
+    if (!pPidlOut)
+        return E_INVALIDARG; // Our pidls are special, the caller must get it from us and not from parsing
+    *pPidlOut = NULL;
+    HRESULT hr;
```

Comment:
Making the code more readable (less cluttered):
```suggestion

    HRESULT hr;
```